pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfdtopologyupdater
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	v1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"sigs.k8s.io/node-feature-discovery/pkg/dumpobject"
+	pb "sigs.k8s.io/node-feature-discovery/pkg/topologyupdater"
+	"sigs.k8s.io/node-feature-discovery/pkg/version"
+)
+
+var (
+	stdoutLogger = log.New(os.Stdout, "", log.LstdFlags)
+	stderrLogger = log.New(os.Stderr, "", log.LstdFlags)
+	nodeName     = os.Getenv("NODE_NAME")
+)
+
+// Command line arguments
+type Args struct {
+	CaFile             string
+	CertFile           string
+	KeyFile            string
+	NoPublish          bool
+	Oneshot            bool
+	Server             string
+	ServerNameOverride string
+}
+
+type NfdTopologyUpdater interface {
+	Update(v1alpha1.ZoneList) error
+}
+
+type nfdTopologyUpdater struct {
+	args       Args
+	clientConn *grpc.ClientConn
+	client     pb.NodeTopologyClient
+	tmPolicy   string
```

Comment:
@marquiz Just want to make sure IIUC. I will split this function as follows:
```

type staticNodeInfo struct {
	args     Args
	tmPolicy string
}

type nfdTopologyUpdater struct {
	clientConn *grpc.ClientConn
	client     pb.NodeTopologyClient
	nodeInfo   *staticNodeInfo
}
```
Is it what you are thinking as well?

Reply:
Hmm, trying to remember what I was after 🧐 I think I was thinking about putting only `tmPolicy` into a separate struct, args wouldn't belong there. In future, if there would be other "static" information that the daemon detects at startup time it would be put in this struct instead of cluttering `nfdTopologyUpdater` with a motley set of node info fields.

I.e.
```
type staticNodeInfo struct {
	tmPolicy string
}

type nfdTopologyUpdater struct {
	nfdclient.NfdBaseClient

	args                Args
	resourcemonitorArgs resourcemonitor.Args
	nodeInfo  *staticNodeInfo
	certWatch *utils.FsWatcher
	client    pb.NodeTopologyClient
	stop      chan struct{} // channel for signaling stop
}
```

Reply:
Ack, updated as per recommendation here.